Src/Support/Google.Apis.Core/Http/HttpClientFactory.cs

```diff
@@ -56,48 +134,50 @@ protected virtual HttpMessageHandler CreateHandler(CreateHttpClientArgs args)
 
             if (!args.GZipEnabled)
             {
                 // Simple: nothing will be decompressing content, so we can just intercept the original handler.
-                return new StreamInterceptionHandler(clientHandler);
+                return new StreamInterceptionHandler(messageHandler);
             }
-            else if (!clientHandler.SupportsAutomaticDecompression)
+            else if (innerHandler?.SupportsAutomaticDecompression != true)
             {
                 // Simple: we have to create our own decompression handler anyway, so there's still just a single chain.
-                var interceptionHandler = new StreamInterceptionHandler(clientHandler);
+                var interceptionHandler = new StreamInterceptionHandler(messageHandler);
                 return new GzipDeflateHandler(interceptionHandler);
             }
             else
             {
                 // Complex: we want to use a simple handler with no interception but with built-in decompression
                 // for requests that wouldn't perform interception anyway, and a longer chain for interception cases.
-                clientHandler.AutomaticDecompression = DecompressionMethods.Deflate | DecompressionMethods.GZip;
+                innerHandler.AutomaticDecompression = DecompressionMethods.Deflate | DecompressionMethods.GZip;
 
                 return new TwoWayDelegatingHandler(
                     // Normal handler (with built-in decompression) when there's no interception.
-                    clientHandler,
+                    messageHandler,
                     // Alternative handler for requests that might be intercepted, and need that interception to happen
                     // before decompression. We need to delegate to a new client handler that *doesn't* 
-                    new GzipDeflateHandler(new StreamInterceptionHandler(CreateAndConfigureClientHandler())),
+                    new GzipDeflateHandler(new StreamInterceptionHandler(CreateAndConfigureClientHandler(out _))),
                     request => StreamInterceptionHandler.GetInterceptorProvider(request) != null);
             }
         }
 
         /// <summary>
-        /// Creates a simple client handler with redirection and compression disabled.
+        /// Creates a message handler with redirection and compression disabled,
+        /// if possible.
         /// </summary>
-        private HttpClientHandler CreateAndConfigureClientHandler()
+        private HttpMessageHandler CreateAndConfigureClientHandler(out HttpClientHandler innerHandler)
         {
-            var handler = CreateClientHandler();
-            if (handler.SupportsRedirectConfiguration)
+            var handler = HttpMessageHandlerFactory();
+            innerHandler = UnderlyingHttpClientHandlerGetter(handler);
+            if (innerHandler?.SupportsRedirectConfiguration == true)
```

Comment:
Hi, my opinion is that someone configured `HttpMessageHandlerFactory` and `UnderlyingHttpClientHandlerGetter`. So the responsibility is on them.

Yes, you can create `HttpMessageHandler` (a `DelegatingHandler`) that wraps existing `HttpClientHandler`.

Reply:
We still need to configure in the cases that we own the inner handler, because that's what we've been doing until now. If we change that behaviour we risk breaking someone.
But for new users I'll add the explicit possibility to allow or not the handler configuration. It can be done implicitly now by returning null for the inner handler, but I agree that might be missed and/or not be enough for covering all cases, for instance, configuring redirects only, but not compession.

Reply:
I'd really like to check what happens with HttpClientFactory - because if HttpClientHandlers are reused or possibly even used concurrently, that could make this whole PR infeasible.